internal/util/csiconfig.go

```diff
@@ -23,52 +23,78 @@ import (
 	"strings"
 )
 
-/*
-Mons returns a comma separated MON list from the csi config for the given clusterID
-Expected JSON structure in the passed in config file is,
-[
-	{
-		"clusterID": "<cluster-id>",
-		"monitors":
-			[
-				"<monitor-value>",
-				"<monitor-value>",
-				...
-			]
-	},
-	...
-]
-*/
-
 // clusterInfo strongly typed JSON spec for the above JSON structure
 type clusterInfo struct {
 	ClusterID string   `json:"clusterID"`
 	Monitors  []string `json:"monitors"`
+	CephFS    struct {
+		SubvolumeGroup string `json:"subvolumeGroup"`
+	} `json:"cephFS"`
 }
 
-func Mons(pathToConfig, clusterID string) (string, error) {
+// Expected JSON structure in the passed in config file is,
+// [
+// 	{
+// 		"clusterID": "<cluster-id>",
+// 		"monitors":
+// 			[
+// 				"<monitor-value>",
+// 				"<monitor-value>",
+// 				...
+// 			],
+//         "cephFS": {
+//           "subvolumeGroup": "<subvolumegroup for cephfs volumes>"
+//         }
+// 	},
+// 	...
+// ]
+func readClusterInfo(pathToConfig, clusterID string) (clusterInfo, error) {
 	var config []clusterInfo
 
 	// #nosec
 	content, err := ioutil.ReadFile(pathToConfig)
 	if err != nil {
 		err = fmt.Errorf("error fetching configuration for cluster ID (%s). (%s)", clusterID, err)
-		return "", err
+		return clusterInfo{}, err
 	}
 
 	err = json.Unmarshal(content, &config)
 	if err != nil {
-		return "", fmt.Errorf("unmarshal failed: %v. raw buffer response: %s",
+		return clusterInfo{}, fmt.Errorf("unmarshal failed: %v. raw buffer response: %s",
 			err, string(content))
 	}
 
 	for _, cluster := range config {
 		if cluster.ClusterID == clusterID {
-			if len(cluster.Monitors) == 0 {
-				return "", fmt.Errorf("empty monitor list for cluster ID (%s) in config", clusterID)
-			}
-			return strings.Join(cluster.Monitors, ","), nil
+			return cluster, nil
 		}
 	}
-	return "", fmt.Errorf("missing configuration for cluster ID (%s)", clusterID)
+
+	return clusterInfo{}, fmt.Errorf("missing configuration for cluster ID (%s)", clusterID)
+}
+
+// Mons returns a comma separated MON list from the csi config for the given clusterID
+func Mons(pathToConfig, clusterID string) (string, error) {
+	cluster, err := readClusterInfo(pathToConfig, clusterID)
+	if err != nil {
+		return "", err
+	}
+
+	if len(cluster.Monitors) == 0 {
+		return "", fmt.Errorf("empty monitor list for cluster ID (%s) in config", clusterID)
+	}
+	return strings.Join(cluster.Monitors, ","), nil
+}
+
+// CephFSSubvolumeGroup returns the subvolumeGroup for CephFS volumes. If not set, it returns the default value "csi"
+func CephFSSubvolumeGroup(pathToConfig, clusterID string) (string, error) {
+	cluster, err := readClusterInfo(pathToConfig, clusterID)
+	if err != nil {
+		return "", err
+	}
+
+	if cluster.CephFS.SubvolumeGroup == "" {
+		return "csi", nil
```

Comment:
Please use a const for `"csi"` and define it somewhere in the top of this file with a  clear comment.

Reply:
Done :-)